pytest_fixtures/satellite_auth.py

```diff
@@ -248,48 +251,70 @@ def enroll_configure_rhsso_external_auth():
 
 
 @pytest.fixture(scope='session')
-def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth):
+def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth, default_sat):
     """register the satellite with RH-SSO Server for single sign-on"""
     client_id = get_rhsso_client_id()
     create_mapper(GROUP_MEMBERSHIP_MAPPER, client_id)
     audience_mapper = copy.deepcopy(AUDIENCE_MAPPER)
     audience_mapper['config']['included.client.audience'] = audience_mapper['config'][
         'included.client.audience'
-    ].format(rhsso_host=settings.server.hostname)
+    ].format(rhsso_host=default_sat)
     create_mapper(audience_mapper, client_id)
     set_the_redirect_uri()
 
 
 @pytest.mark.external_auth
 @pytest.fixture(scope='session')
-def enroll_idm_and_configure_external_auth():
+def enroll_idm_and_configure_external_auth(default_sat):
     """Enroll the Satellite6 Server to an IDM Server."""
     run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
-
     run_command(
         cmd=f'echo {settings.ipa.password} | kinit admin',
         hostname=settings.ipa.hostname,
     )
-    result = run_command(
-        cmd=f'ipa host-add --random {settings.server.hostname}',
-        hostname=settings.ipa.hostname,
-    )
+    output = ssh.command(f'ipa host-find {default_sat.hostname}')
+    if output.status != 0:
+        result = run_command(
+            cmd=f'ipa host-add --random {default_sat.hostname}',
+            hostname=settings.ipa.hostname,
+        )
+        for line in result:
+            if 'Random password' in line:
+                _, password = line.split(': ', 2)
+                break
+        ssh.command(
+            cmd=f'ipa service-add HTTP/{default_sat.hostname}',
+            hostname=settings.ipa.hostname,
+        )
+        _, domain = settings.ipa.hostname.split('.', 1)
+        result = ssh.command(
+            cmd=f"ipa-client-install --password '{password}' "
```

Comment:
`password` isn't set, resulting in an exception here.

This suggests the `if 'Random password' in line` check above is never hit, so the variable is never named in this scope.

Reply:
hm, the issue was that the iteration went sign by sign, now it should be looking at lines